homeassistant/components/myq/config_flow.py

```diff
@@ -87,10 +94,41 @@ async def async_step_import(self, user_input):
         self._abort_if_unique_id_configured()
         return await self.async_step_user(user_input)
 
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
 
 class CannotConnect(exceptions.HomeAssistantError):
     """Error to indicate we cannot connect."""
 
 
 class InvalidAuth(exceptions.HomeAssistantError):
     """Error to indicate there is invalid auth."""
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for Harmony."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Handle options flow."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_USERAGENT,
```

Comment:
I don't think user agent should be an option for the user. If that is needed something isn't right with how the integration is working against the API.

Reply:
Hey Martin,

The problem has been that the user agent has been getting blocked resulting in authentication errors. We then change the user agent again to something that then works for a while before it is then blocked again.
Last time a random user agent was implemented and then just this past week it was being blocked. 

Each time it is blocked, the integration stops working until the integration is fixed and then the change released in a new Home Assistant version. And we just keep on doing this. 
The underlying integration has now been updated to also accept a custom user agent and thus this PR to able to leverage that so that folks can provide a new user agent that then works when current one is blocked instead of having to wait.

Reply:
The current issue with myQ **intermittent** connectivity is apparently not related to specific user agent headers being blocked.   Connections to the legacy myQ V5 infrastructure do work for a short time, get dropped due to "500-server" errors, then reconnect repeatedly in short intervals.

The original agent blocking issue was successfully addressed with the update to generate a random agent.   #44348 should have been closed as no longer needed.     In the latest pymyq 2.0.13 update in 2021.1.1, this random agent was removed and "pymq" is now the default user agent with no non-hacking way to change it unless this PR is approved.  IMO the random agent should have been retained.

The focus now needs to be on rewriting the myQ integration to be compatible with the new myQ v6 architecture as described in https://github.com/arraylabs/pymyq/issues/60.   Simply changing the user-agent in the config flow is unlikely to solve anything (unless the now default "pymq" is being blocked).   Please provide evidence connections are in fact stable with a custom agent for as long as myQ v5 remains operational.

@bdraco - do you concur?

Reply:
I was getting Error 400 with the random agent. Changing to pymyq resulted in it being able to authenticate again. Error 500 is still sometimes showing up.
I am not the only one who had Error 400 with the random user agent and then ended up being resolved with going to pymyq.

Reply:
Thanks for pointing out issue  59.    That should have been a key info link in this PR as to what it fixes.  I didn't know that the random agent was causing auth failures.   I had not seen that occur for my two separate myQ garage openers.   I had assumed this update was in response to the 500-server errors.    

It's pretty bad - in the past 24 hours, my logs show 180 myQ 500-server connection drops/reconnects - on average every 8 minutes.    The custom agent won't solve this, but at least it should help those who have auth failures to at least connect and then complain about the new issue!

Reply:
Home Assistant should not actively be trying to work around measures from API providers to stop us from using their API. We should work together with API providers start a conflict.

It sounds like myQ doesn't want us to use their API like we currently do. It would be better to try and solve this underlying issue.

We won't accept this PR.

Reply:
There are many HASS integrations based on non-public API's that are reversed engineered.  It's hard to say which of those  are perfectly OK - or frowned on, subject to being blocked.   

I see from comments in the forum that many people have personally contacted Chamberlain requesting a public API but that is very unlikely to happen.    Instead, Chamberlain has an established "Works with MyQ" partnership program where businesses can sign up to participate.

Would Nabu Casa be interested in signing up?    The form is at the bottom of this link...
https://www.myq.com/works-with-myq